Backend updated

ritishab0209 opened the pull request:
Please look into the changes made and merge it

ritishab0209 committed:
Backend updated

nightmare-tech commented:
Please give me an overview of what you've changed/updated/improved.

nightmare-tech commented on logs/xvba_debug.log:
Shouldn't this file be in gitignore?

nightmare-tech replied:
Put the logs folder in gitignore

ritishab0209 replied:
Done

ritishab0209 commented:
Yes, so I have used bcrypt to secure passwords in a better way
Moved the Encryption key to an environment variable for better security
Replaced aes-ecb with aes-cbc for better security
These are the important changes I made, hoping that these changes will get
approved

On Fri, 11 Oct, 2024, 9:55 am Arpit Gupta, ***@***.***> wrote:

> Please give me an overview of what you've changed/updated/improved.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nightmare-tech/secure_storage/pull/5#issuecomment-2406530102>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BFUQC5SQZ2DS7HN6MXAEJNLZ25HMVAVCNFSM6AAAAABPYC3BQOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIMBWGUZTAMJQGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

nightmare-tech commented on app.py:
```diff
@@ -6,16 +6,19 @@
-# Set the encryption key (ideally, this should be fetched securely from a key management service)
-encryption_key = b'0123456789abcdef0123456789abcdef'
+# Fetch the encryption key securely from an environment variable
+encryption_key = os.getenv('ENCRYPTION_KEY', '0123456789abcdef0123456789abcdef').encode()
```
You should add a warning when the fallback key is used instead of the environment key. This would only happen when the environment variable is not set.

how about this? :
```python
def get_encryption_key():
    key = os.getenv('ENCRYPTION_KEY')
    if key is None:
        print("Warning: Using default encryption key. This is insecure for production use.")
        return b'0123456789abcdef0123456789abcdef'
    return key.encode()

encryption_key = get_encryption_key()
```

nightmare-tech commented on app.py:
```diff
@@ -38,53 +41,44 @@ def init_db():
-    # Remove padding from the plaintext
     unpadder = padding.PKCS7(algorithms.AES.block_size).unpadder()
-    unpadded_plaintext = unpadder.update(plaintext) + unpadder.finalize()
+    unpadded_plaintext = unpadder.update(plaintext) + unpadded_plaintext.finalize()
```
how does this work? isnt  `unpadded_plaintext` not initialised?

nightmare-tech reviewed (changes requested):
Great!! Here are some things to address:
- include bcrypt in requirements.txt
- other comments in files

ritishab0209 commented:
I'll look through it and make all the necessary changes

ritishab0209 committed:
Updated

ritishab0209 commented:
Please review it, I have made the said changes

nightmare-tech commented:
I'm no expert on cryptography but a Google search supports my statement

ritishab0209 committed:
Update .gitignore

ritishab0209 commented:
I would say bcrypt is a better option for the safe storage of passwords

On Fri, 11 Oct, 2024, 10:35 pm Arpit Gupta, ***@***.***>
wrote:

> I'm no expert on cryptography but a Google search supports my statement
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nightmare-tech/secure_storage/pull/5#issuecomment-2407810690>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BFUQC5XBFE7BIPLXN2ANCM3Z3AAOLAVCNFSM6AAAAABPYC3BQOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIMBXHAYTANRZGA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

nightmare-tech commented:
Let me know if this shows up as a contribution in your hacktoberfest account.

ritishab0209 commented:
Yes it shows as accepted, thanks

nightmare-tech commented:
Solves #4
